File: exe5.py
class Animal:
    def som(self):
        print("O animal faz um som.")

class Cachorro(Animal):
    def som(self):
        print("O cachorro late: Au Au!")

class Gato(Animal):
    def som(self):
        print("O gato mia: Miau!")

def emitir_som_dos_animais(animais):
    for animal in animais:
        animal.som() 
if __name__ == "__main__":
    animais = [Cachorro(), Gato()]

    emitir_som_dos_animais(animais)

File: exe5.java
import java.util.ArrayList;

class Animal {
    public void som() {
        System.out.println("O animal faz um som.");
    }
}

class Cachorro extends Animal {
    @Override
    public void som() {
        System.out.println("O cachorro late: Au Au!");
    }
}

class Gato extends Animal {
    @Override
    public void som() {
        System.out.println("O gato mia: Miau!");
    }
}

public class exe5 {
    public static void emitirSomDosAnimais(ArrayList<Animal> animais) {
        for (Animal animal : animais) {
            animal.som();  
        }
    }

    public static void main(String[] args) {
        ArrayList<Animal> animais = new ArrayList<>();
        animais.add(new Cachorro());
        animais.add(new Gato());

        emitirSomDosAnimais(animais);  
    }
}
